src/signal_map.rs

```diff
@@ -165,6 +180,36 @@ pub type BoxSignalMap<'a, Key, Value> = Pin<Box<dyn SignalMap<Key = Key, Value =
 pub type LocalBoxSignalMap<'a, Key, Value> = Pin<Box<dyn SignalMap<Key = Key, Value = Value> + 'a>>;
 
 
+#[derive(Debug)]
+#[must_use = "SignalMaps do nothing unless polled"]
+pub struct Always<K, V> {
+    entries: Option<Vec<(K, V)>>,
+}
+
+impl<K, V> Unpin for Always<K, V> {}
+
+impl<K, V> SignalMap for Always<K, V> {
+    type Key = K;
+    type Value = V;
+
+    #[inline]
+    fn poll_map_change(mut self: Pin<&mut Self>, _cx: &mut Context) -> Poll<Option<MapDiff<Self::Key, Self::Value>>> {
+        Poll::Ready(self.entries.take().map(|entries| MapDiff::Replace { entries }))
+    }
+}
+
+/// Converts a `Vec<(K, V)>` into a `SignalMap<Key = K, Value = V>`
+///
+/// This has no performance cost.
+#[inline]
+// TODO `Vec<(K, V)>` -> `BTreeMap<K, V>`?
+pub fn always<K, V>(entries: Vec<(K, V)>) -> Always<K, V> {
```

Comment:
I think this should accept a `BTreeMap` instead. That is more consistent, and it also guarantees that each key only exists once.

Reply:
Changed to `BTreeMap` in the latest commit.

I wasn't sure whether to choose `BTreeMap` or `Vec<(K,V)>` because `entries` in `MapDiff::Replace { entries }` are defined as `Vec`.  What's the reason for using `Vec` instead of `BTreeMap` in that `MapDiff`?

Reply:
Mostly because `MapDiff` is an internal detail and using `Vec` is more efficient. But for user-facing APIs it's good for it to be an actual dictionary.